src/zimscraperlib/zim/creator.py

```diff
@@ -110,26 +102,114 @@ def __init__(
                 else compression
             )
 
-        if metadata:
-            self.metadata = metadata
-
         self.workaround_nocancel = workaround_nocancel
         self.ignore_duplicates = ignore_duplicates
 
+    def set_mainpath(self, main_path):
+        super().set_mainpath(main_path)
+
     def start(self):
         super().__enter__()
 
-        if getattr(self, "main_path", None):
-            self.set_mainpath(self.main_path)
+        if not self._all_metadata_is_set:
+            raise ValueError("Mandatory metadata are not all set.")
+        self.add_metadata(**self._metadata)
+        self.add_default_illustration(self._metadata["Illustration_48x48@1"])
 
-        if getattr(self, "metadata", None):
-            self.update_metadata(**self.metadata)
         return self
 
-    def update_metadata(self, **kwargs):
+    def add_metadata(self, **kwargs) -> None:
+        """
+        Use this function to add additional Metadata, please note that the added Metadata should not be duplicated.
+        """
         if kwargs:
-            for name, value in kwargs.items():
-                self.add_metadata(name, value)
+            for name, value in self._metadata.items():
+                # Illustration needs to be added at the end of start()
+                if name != "Illustration_48x48@1" and value:
```

Comment:
It might be a better note to store Illustration in a separate variable?

Reply:
What for? If we get rid (`delete`) `self._metadata` in `start()` it's fine I think